### PR TITLE
add function to get title from server for teatime.html

### PR DESCRIPTION
**Pull request by trinitydonohugh**

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -29,3 +29,14 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerHTML = gif;
 }
+
+/**
+ * gets title from server
+ */
+function getTitle() {
+  console.log('in getTitle() function');
```
Comment (pengxianzhe) [inline portfolio/src/main/webapp/script.js:37]: This line looks like testing code, is it intended?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -29,3 +29,14 @@ function addRandomGreeting() {
...
+  fetch('/data').then(response => response.text()).then((titleText) => {
+    console.log('printing content');
```
Comment (pengxianzhe) [inline portfolio/src/main/webapp/script.js:39]: Ditto

File: portfolio/src/main/webapp/teatime.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js" integrity="sha384-ZMP7rVo3mIykV+2+9J3UJ46jBk0WLaUAdn689aCwoqbBJiSnjAK/l8WvCWPIPm49" crossorigin="anonymous"></script>
+    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js" integrity="sha384-ChfqqxuZUCnJSK3+MXmPNIyE6ZbWh2IMqE241rYiqJxyMiZ6OW/JmZQ5stwEULTy" crossorigin="anonymous"></script>
+  </head>
+  <body onload="getTitle()">
+    <div class="container">
+      <div id="title-text"></div>
+    </div>
+  </body>
+</html>
```
Comment (pengxianzhe) [inline portfolio/src/main/webapp/teatime.html:20]: nit: no new line at EOF

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (trinitydonohugh) [reply portfolio/src/main/webapp/script.js]: Yes, so sorry, I forgot to delete them. I have deleted them in the new commit below! :)

File: portfolio/src/main/webapp/teatime.html (hunk 3 above)
Comment (trinitydonohugh) [reply portfolio/src/main/webapp/teatime.html]: I've added a new line to the new commit! So sorry.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -14,21 +14,42 @@
...
+  }
 
-    response.setContentType("text/html;");
-    response.getWriter().println(titleText);
+  /**
+   * Converts the messages list instance into a JSON string using the Gson library. 
+   */
+  private String convertToJson(List<String> messages) {
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: There is no need to declare a variable for this string, you can just return it right away.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -14,19 +14,42 @@
...
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns messages/ comments. */
```
Comment (isaacwdavis) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:26]: messages/ comments -> messages/comments

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (isaacwdavis) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: "return new Gson().toJson(messages);" if you really want to make it concise, but "return gson.toJson(messages);" also works

Comment (trinitydonohugh) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: I've added that now so it'll be concise in my next commit! :) Thank you!!

**Pull request merged**